### PR TITLE
Modify the log level of functions and move the java_instance_log4j2.xml file from the jar package to the conf directory

### PR DESCRIPTION
**Pull request by g0715158**
Motivation

The current functions log configuration is in the jar package, and cannot be dynamically customized..

The target of this PR is making the functions log configuration file to the conf directory, which can be customized.

Modifications
1.Add configuration file directory to get from env

Verifying this change
 Make sure that the change passes the CI checks.

This change added tests

Does this pull request potentially affect one of the following parts:

If yes was chosen, please highlight the changes
•Dependencies (does it add or upgrade a dependency): (no)
•The public API: (no)
•The schema: (no)
•The default values of configurations: (yes)
•The wire protocol: (no)
•The rest endpoints: (no)
•The admin cli options: (no)
•Anything that affects deployment: (don't know)

Documentation
•Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Comment by Jennifer88huang** [comment]
@g0715158 Thank you very much for your contribution.
You can delete unnecessary info in PR description, and keep key info, like "Motivation" and "Modification". Here is an [example](https://github.com/apache/pulsar/pull/7552).

**Comment by jiazhai** [comment]
@g0715158 Would you please also put a copy of the file `java_instance_log4j2.xml` into `conf` directory?

**Comment by g0715158** [comment]
OK。think you

发自我的iPhone

File: pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/process/ProcessRuntime.java [hunk 1]
```diff
@@ -106,6 +106,10 @@
         switch (instanceConfig.getFunctionDetails().getRuntime()) {
             case JAVA:
                 logConfigFile = "java_instance_log4j2.xml";
+                String palsarHome = System.getenv("PULSAR_HOME");
```
Comment (cdbartholomew) [inline pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/process/ProcessRuntime.java:109]: You have a typo in the variable name: `palsarHome` should be `pulsarHome`.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/NamespaceStatsAggregator.java [hunk 2]
```diff
@@ -127,7 +127,12 @@ private static void getTopicStats(Topic topic, TopicStats stats, boolean include
                 stats.throughputIn += producer.getStats().msgThroughputIn;
             }
         });
-
+        topic.getSubscriptions().values().forEach(sub->{
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/NamespaceStatsAggregator.java:130]: This change seems unrelated, please create another pr foe this

File: pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/process/ProcessRuntime.java [hunk 3]
```diff
@@ -106,6 +106,10 @@
...
+                if(palsarHome!=null&&!palsarHome.trim().isEmpty()){
+                    logConfigFile = palsarHome+ "/conf/java_instance_log4j2.xml";
```
Comment (eolivelli) [inline pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/process/ProcessRuntime.java:111]: Can we use a more user friendly name?
Like pulsar_functions_log4j2.xml

**Review by eolivelli — CHANGES_REQUESTED** [review]
I really love this change.
I was going to do something similar as currently it is not possible to change log level for functions.

Also the log4j file here pollutes the log4j conf when running tests, making it quite impossible to change log level while working on tests

File: pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/process/ProcessRuntime.java [hunk 4]
```diff
@@ -106,6 +106,10 @@
         switch (instanceConfig.getFunctionDetails().getRuntime()) {
             case JAVA:
                 logConfigFile = "java_instance_log4j2.xml";
+                String palsarHome = System.getenv("PULSAR_HOME");
+                if(palsarHome!=null&&!palsarHome.trim().isEmpty()){
```
Comment (vaihtovirta) [inline pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/process/ProcessRuntime.java:110]: - I believe this check is redundant since `PULSAR_HOME` is set in [bin/pulsar](https://github.com/apache/pulsar/blob/313390d86f7ff81124f75181bdd5bbd2b2be7fca/bin/pulsar#L22)
- Instead of having a hardcoded path in the `conf` folder, can we read from a separate environment variable, e.g `PULSAR_FUNCTIONS_LOG_CONF`?
Here is how it works already for broker logs with [`PULSAR_LOG_CONF`](https://github.com/apache/pulsar/blob/313390d86f7ff81124f75181bdd5bbd2b2be7fca/bin/pulsar#L160).

File: pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/process/ProcessRuntime.java (hunk 1 above)
Comment (jerrypeng) [inline pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/process/ProcessRuntime.java:109]: Please also move this line under the subsequent "if" statement as it is only used when the "if" conditional resolves to true

**Comment by eolivelli** [comment]
probably this branch is superseded by d757db19c6553ab63674464310357b57fa906903 ?

**Comment by Anonymitaet** [comment]
@g0715158 thanks for your code work. Does this need to update docs? If so, could you please help add docs accordingly? Then I can help review, thanks

**Comment by codelipenghui** [comment]
Any update on this PR @g0715158? I will move it to milestone 2.9.0 first, feel free to left comments if want to move it back to 2.8.0 release.

**Comment by tisonkun** [comment]
Closed by https://github.com/apache/pulsar/pull/10389.
